[markup] 웹 학습 사이트 상세, 작성, 수정, 리스트 페이지 마크업

EliteZer0 opened the pull request:
## #️⃣ Issue Number

closes #95
closes #96 
closes #97 

## 📝 요약(Summary)

- API에서 generationId 타입에 null을 허용하도록 변경
- WebSiteCard 컴포넌트에서 북마크 관련 props와 UI 제거
- WebRecommendCards 컴포넌트에서 웹 추천 검색 기능 추가 및 불필요한 북마크 코드 제거
- 콘텐츠 삭제 후 모달창이 닫히지 않는 버그 수정
- 웹 추천 생성 시 등록된 URL일 경우 에러메세지 모달 추가
- 웹 추천 수정 페이지 구현

## 🛠️ PR 유형

어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [x] 버그 수정
- [x] CSS 등 사용자 UI 디자인 변경
- [x] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## 💬 공유사항 to 리뷰어

## 📸스크린샷 (선택)

## ✅ PR Checklist

PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

EliteZer0 committed:
[fix] generationId 타입 수정

generationId 타입에 null을 허용하도록 변경

EliteZer0 committed:
[refactor] 북마크 관련 기능 제거

WebSiteCard 컴포넌트에서 북마크 관련 props와 UI 제거

EliteZer0 committed:
[refactor] 웹 추천 검색 기능 추가 및 불필요한 북마크 코드 제거

- 웹 추천 검색을 위한 searchTitle state 추가
- API 호출 시 searchTitle 파라미터 추가
- 불필요한 북마크 관련 props 제거
- 날짜 표시 필드를 updatedAt에서 createdAt으로 변경

EliteZer0 committed:
[fix] 콘텐츠 삭제 후 모달창이 닫히지 않는 버그 수정

- 콘텐츠 삭제 완료 후 closeModal() 함수 호출하여 모달창 닫기 추가

EliteZer0 committed:
feat: 웹 추천 생성 시 등록된 URL 에러메세지 표시 기능 구현

- 이미 등록된 URL인 경우 별도의 에러 메시지 표시

EliteZer0 committed:
[fix] 웹 추천 상세 페이지의 contentType 오타 수정

- contentType 값을 'web_guide'에서 'web_recommend'로 수정

EliteZer0 committed:
[feat] 웹 추천 수정 페이지 구현

- 웹 추천 수정 기능 구현
- 기존 데이터 조회 및 폼 초기값 설정
- 제목, URL, 내용 수정 기능 추가
- 유효성 검사 및 에러 처리
- 취소/제출 핸들러 구현

clapsheep commented on src/pages/webRecommend/WebRecommendUpdate.tsx:
```diff
@@ -1,5 +1,131 @@
-    return <div>웹 추천 수정</div>;
+    const { pathname } = useLocation();
+    const { recommendId } = useParams();
+    const { data } = useQuery({
```
디테일 페이지에서 사용하는 useContentDetail에 전달하는 queryKey가 같다면 이렇게 직접 useQuery로 구현하지 않고 그 훅에 같은 쿼리키를 전달하면 동일한 캐싱된 데이터를 얻어오고 사용할 수 있었지 않았을까 싶은데 그렇게 해보는 방법은 어떨까요? 코드가 많이 짧아질 겁니다! 대신 useEffect로 해당 데이터에 해당하는 state를 미리 채우는 로직은 필요할 거에요

EliteZer0 committed:
Merge branch 'develop' into feat/EliteZer0

EliteZer0 committed:
[refactor] useContentDetail 훅을 사용하여 데이터 페칭 로직 통합

- 개별적인 useQuery 대신 useContentDetail 훅을 사용하여 데이터 페칭
- 상태 초기화 로직을 useEffect로 분리하여 데이터 로드 시점에 맞춰 업데이트
